var/spack/repos/builtin/packages/charliecloud/package.py

```diff
@@ -28,7 +29,8 @@ class Charliecloud(AutotoolsPackage):
     depends_on('libtool',  type='build')
 
     depends_on('python@3.5:',    type='run')
-    depends_on('py-lark-parser', type='run')
+    # Version 0.25+ bundle the preferred lark version.
+    depends_on('py-lark-parser', type='run', when='@:0.24')
```

Comment:
(comment/question: no change needed) normally vendored dependencies can be problematic: if another package were to depend on `charliecloud` and `py-lark-parser`, there could be a consistency issue. However, I think `charliecloud` is always deployed as a top-level package (i.e. no dependents) so it shouldn't generally have that problem (although it's still possible that an environment which installs `charliecloud` and another package which depends on `py-lark-parser` could get `PYTHONPATH` conflicts).

Ideally then, it would be possible to configure `charliecloud` to use an external `py-lark-parser` - do you know if that is possible?